File: app/src/main/java/com/my/televip/obfuscate/struct/CustomResolverOverlay.java
package com.my.televip.obfuscate.struct;

import java.util.HashMap;
import java.util.Map;

public class CustomResolverOverlay {

    private static final Map<String, String> classMap = new HashMap<>();
    private static final Map<String, String> fieldMap = new HashMap<>();
    private static final Map<String, String> methodMap = new HashMap<>();

    public static void putClassOverride(String originalName, String resolvedName) {
        classMap.put(originalName, resolvedName);
    }

    public static void putFieldOverride(String className, String fieldName, String resolvedFieldName) {
        fieldMap.put(className + "#" + fieldName, resolvedFieldName);
    }

    public static void putMethodOverride(String className, String methodName, String resolvedMethodName) {
        methodMap.put(className + "#" + methodName, resolvedMethodName);
    }

    public static boolean hasClassOverride(String originalName) {
        return classMap.containsKey(originalName);
    }

    public static String getClassOverride(String originalName) {
        return classMap.get(originalName);
    }

    public static boolean hasFieldOverride(String className, String fieldName) {
        return fieldMap.containsKey(className + "#" + fieldName);
    }

    public static String getFieldOverride(String className, String fieldName) {
        return fieldMap.get(className + "#" + fieldName);
    }

    public static boolean hasMethodOverride(String className, String methodName) {
        return methodMap.containsKey(className + "#" + methodName);
    }

    public static String getMethodOverride(String className, String methodName) {
        return methodMap.get(className + "#" + methodName);
    }
}

File: app/src/main/java/com/my/televip/obfuscate/struct/CustomResolverOverlay.kt
package com.my.televip.obfuscate.struct

import java.util.concurrent.ConcurrentHashMap

object CustomResolverOverlay {

    private val classMap = ConcurrentHashMap<String, String>()
    private val fieldMap = ConcurrentHashMap<String, String>()
    private val methodMap = ConcurrentHashMap<String, String>()

    @JvmStatic
    fun putClassOverride(originalName: String, resolvedName: String) {
        classMap[originalName] = resolvedName
    }

    @JvmStatic
    fun putFieldOverride(className: String, fieldName: String, resolvedFieldName: String) {
        fieldMap["$className#$fieldName"] = resolvedFieldName
    }

    @JvmStatic
    fun putMethodOverride(className: String, methodName: String, resolvedMethodName: String) {
        methodMap["$className#$methodName"] = resolvedMethodName
    }

    @JvmStatic
    fun hasClassOverride(originalName: String): Boolean {
        return classMap.containsKey(originalName)
    }

    @JvmStatic
    fun getClassOverride(originalName: String): String? {
        return classMap[originalName]
    }

    @JvmStatic
    fun hasFieldOverride(className: String, fieldName: String): Boolean {
        return fieldMap.containsKey("$className#$fieldName")
    }

    @JvmStatic
    fun getFieldOverride(className: String, fieldName: String): String? {
        return fieldMap["$className#$fieldName"]
    }

    @JvmStatic
    fun hasMethodOverride(className: String, methodName: String): Boolean {
        return methodMap.containsKey("$className#$methodName")
    }

    @JvmStatic
    fun getMethodOverride(className: String, methodName: String): String? {
        return methodMap["$className#$methodName"]
    }
}
